### PR TITLE
Shared feed wording and navigation tweaks

### PR DESCRIPTION
**Pull request by brianfleming**
## Description

- Updated wording for shared feed details page for noting that the updated date was for settings
- The side navigation would "flicker" when visiting just the settings area because of a less optimal regex which had the page list briefly changing to the tipline list and then back to the settings list
- Moved a few of the side navigation files to the drawer folder for better organization
- Removed some unnecessary mui styles in favor of css file

References: [CV2-4232]

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

### BEFORE side navigation "flicker"

https://github.com/meedan/check-web/assets/418001/5a394633-150d-4d99-b61d-844a023703a6

### AFTER Last updated
Screenshot 2024-03-21 at 11 00 11 AM

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

[CV2-4232]: https://meedan.atlassian.net/browse/CV2-4232?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/components/drawer/DrawerRail.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ const messages = defineMessages({
 
 const DrawerRail = (props) => {
   const testPath = window.location.pathname;
-  const isSettingsPage = /\/settings\/[a-zA-Z0-9]+/.test(testPath);
+  const isSettingsPage = /^\/[^/]+\/settings?($|\/)/.test(testPath);
```
Comment (dariusk) [inline src/app/components/drawer/DrawerRail.js:57]: What's this regular expression supposed to do? When I run something like

```
"https://checkmedia.org/check-testing/settings/workspace".match(/^\/[^/]+\/settings?($|\/)/) 
```

I get no match, which leads me to believe this doesn't work for testing if we are on the settings page at all. It looks like the regex is actually searching for `/` and then at least one of something that's not `/` and then `/` and then `settings` and then either the end of a string or another `/`. See here:

https://regex101.com/r/g80Mip/1

Maybe you meant something like `/[^/]+\/settings?/`? I'm just not sure what's going on here and it doesn't seem to match the trivial string of a settings url.

**Review by dariusk — CHANGES_REQUESTED** [review]

File: src/app/components/drawer/DrawerRail.js (hunk 1 above)
Comment (brianfleming) [reply src/app/components/drawer/DrawerRail.js]: @dariusk I was just trying to figure out why this link was behaving differently than the others, so I copied the shared feeds line from that same file:
`const isFeedPage = /^\/[^/]+\/feed(s)?($|\/)/.test(testPath);`
but perhaps that was overthinking/underthinking it.

I replaced it with your suggestion and it seems to work, take a look again

**Review by danielevalverde — APPROVED** [review]

**Comment by danielevalverde** [comment]
:LGTM :heart_eyes_cat: !!

**Comment by brianfleming** [comment]
just a reminder @dariusk since you requested changes merging is blocked until you re-review. Thanks

**Review by dariusk — APPROVED** [review]
I approve my own code!!

**Pull request merged**